seed/analysis_pipelines/better.py

```diff
@@ -0,0 +1,448 @@
+        for meter_reading in meter.meter_readings.all():
+            if meter_reading.reading is None:
+                errors.append(f'{meter}: MeterReading starting at {meter_reading.start_time} has no reading value')
+    if errors:
+        return None, errors
+
+    # clean inputs
+    # BETTER will default if eGRIDRegion is not explicitly set
+    try:
+        eGRIDRegion = property_state.extra_data['eGRIDRegion']
+    except KeyError:
+        eGRIDRegion = ""
+
+    gross_floor_area = str(int(property_state.gross_floor_area.magnitude))
+
+    XSI_URI = 'http://www.w3.org/2001/XMLSchema-instance'
+    nsmap = {
+        'xsi': XSI_URI,
+    }
+    nsmap.update(NAMESPACES)
+    E = ElementMaker(
+        namespace=BUILDINGSYNC_URI,
+        nsmap=nsmap
+    )
+
+    doc = (
+        E.BuildingSync(
+            {
+                etree.QName(XSI_URI,
+                            'schemaLocation'): 'http://buildingsync.net/schemas/bedes-auc/2019 https://raw.github.com/BuildingSync/schema/v2.2.0/BuildingSync.xsd',
+                'version': '2.2.0'
+            },
+            E.Facilities(
+                E.Facility(
+                    {'ID': 'Facility-1'},
+                    E.Sites(
+                        E.Site(
+                            {'ID': 'Site-1'},
+                            E.Buildings(
+                                E.Building(
+                                    {'ID': 'Building-1'},
+                                    E.PremisesName(property_state.property_name),
+                                    E.OccupancyClassification("Office"),
+                                    E.eGRIDRegionCode(eGRIDRegion),
+                                    E.Address(
+                                        E.City(property_state.city),
+                                        E.State(property_state.state),
+                                        E.PostalCode(property_state.postal_code)
+                                    ),
+                                    E.FloorAreas(
+                                        E.FloorArea(
+                                            E.FloorAreaType("Gross"),
+                                            E.FloorAreaValue(gross_floor_area)
+                                        )
+                                    ),
+                                    E.PremisesIdentifiers(
+                                        E.PremisesIdentifier(
+                                            E.IdentifierLabel('Custom'),
+                                            E.IdentifierCustomName(PREMISES_ID_NAME),
+                                            E.IdentifierValue(str(analysis_property_view.id)),
+                                        )
+                                    ),
+                                    E.Longitude(str(analysis_property_view.property_state.longitude)),
+                                    E.Latitude(str(analysis_property_view.property_state.latitude)),
```

Comment:
should this be replaced with the city, state, zip?

Reply:
geo lat/long are optional BETTER inputs so maybe I could throw in a check to use the values if the property has them and exclude them otherwise.

Reply:
ah gotcha, this seems good then 👍